src/KeyringController.ts

```diff
@@ -930,6 +930,7 @@ class KeyringController extends EventEmitter {
 
     if (
       this.password &&
+      (!this.#cacheEncryptionKey || !encryptionKey) &&
```

Comment:
This ensures that we don't do the whole `persistAllKeyrings` operation for nothing, as the vault will not get updated if we reuse the cached encryption key